Codigo/modNetwork.bas

```diff
@@ -122,89 +131,82 @@ End Function
+    Call TraceError(Err.Number, Err.Description, "modNetwork.Kick", Erl)
 End Sub
+
 Private Sub OnServerConnect(ByVal Connection As Long, ByVal Address As String)
-On Error GoTo OnServerConnect_Err:
-    NewFrameConnections = NewFrameConnections + 1
-    If IsFeatureEnabled("debug_connections") Then
-        Call AddLogToCircularBuffer("OnServerConnect connecting new user on id: " & Connection & " ip: " & Address)
-    End If
-    If Mapping(Connection).ArrayIndex > 0 Then
-        Call TraceError(Err.Number, Err.Description, "OnServerConnect Mapping(Connection) > 0, connection: " & Connection & " value: " & Mapping(Connection).ArrayIndex & ", is valid: " & IsValidUserRef(Mapping(Connection)), Erl)
-    End If
-    If Connection <= MaxUsers Then
-        Dim FreeUser As Long
-        FreeUser = NextOpenUser()
-        If IsFeatureEnabled("debug_id_assign") Then
-            Call LogError("Assign userId: " & FreeUser & " to connection: " & Connection)
-        End If
-        If FreeUser < 0 Then
-            If IsFeatureEnabled("debug_connections") Then
-                Call LogError("Failed to find slot for new user, connection: " & Connection & " LastUser: " & LastUser)
-            End If
-            Call Kick(Connection, "El server se encuentra lleno en este momento. Disculpe las molestias ocasionadas.")
-            Exit Sub
+    On Error GoTo OnServerConnect_Err:
+100     NewFrameConnections = NewFrameConnections + 1
+102     If IsFeatureEnabled("debug_connections") Then
+104         Call AddLogToCircularBuffer("OnServerConnect connecting new user on id: " & Connection & " ip: " & Address)
         End If
-        If UserList(FreeUser).InUse Then
-           Call LogError("Trying to use an user slot marked as in use! slot: " & FreeUser)
-           FreeUser = NextOpenUser()
+106     If Mapping(Connection).UserRef.ArrayIndex > 0 Then
+108         Call TraceError(Err.Number, Err.Description, "OnServerConnect Mapping(Connection).UserRef.ArrayIndex > 0, connection: " & Connection & " value: " & Mapping(Connection).UserRef.ArrayIndex & ", is valid: " & IsValidUserRef(Mapping(Connection).UserRef), Erl)
         End If
-        UserList(FreeUser).ConnIDValida = True
-        UserList(FreeUser).IP = Address
-        UserList(FreeUser).ConnID = Connection
-        UserList(FreeUser).Counters.OnConnectTimestamp = GetTickCount()
         
-        If FreeUser >= LastUser Then LastUser = FreeUser
-        Debug.Assert Not IsValidUserRef(Mapping(Connection))
-        If Not SetUserRef(Mapping(Connection), FreeUser) Then
-            Call TraceError(Err.Number, Err.Description, "OnServerConnect failed to map connection (" & Connection & ") to user: " & FreeUser, Erl)
+110     If Connection <= MaxActiveConnections Then
+112         If IsValidUserRef(Mapping(Connection).UserRef) Then
+114             LogError ("opening a new connection: " & Connection & " to a connection mapped to a user " & Mapping(Connection).UserRef.ArrayIndex)
```

Comment:
should we abort and do `exit sub` ?

Reply:
not really, we should log the error and keep working, if we are here it means that aurora already assigned the connection to someone else and we failed to the the required cleanup